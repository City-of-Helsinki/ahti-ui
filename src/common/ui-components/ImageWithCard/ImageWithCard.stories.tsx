import React from 'react';

import ImageWithCard from './ImageWithCard';

export default {
  title: 'ImageWithCard',
  component: ImageWithCard,
};

export const IndexPageExample = () => {
  return (
    <ImageWithCard
      src="https://scstylecaster.files.wordpress.com/2018/01/beyonce.jpg"
      className=""
    >
      <h1>Hello, this is a demo </h1>
    </ImageWithCard>
  );
};
