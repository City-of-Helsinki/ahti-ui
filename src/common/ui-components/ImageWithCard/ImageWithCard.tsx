import React from 'react';

export type ImageWithCardProps = React.PropsWithChildren<{
  src: string;
  className?: string;
}>;

const ImageWithCard: React.FC<ImageWithCardProps> = ({
  src,
  children,
  className
}) => {
  return (
    <div style={{ backgroundImage: `url(${src})` }} className={className}>
      {children}
    </div>
  );
};

export default ImageWithCard;
