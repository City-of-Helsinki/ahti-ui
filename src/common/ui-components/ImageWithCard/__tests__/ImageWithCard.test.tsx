import React from 'react';
import { shallow } from 'enzyme';

import ImageWithCard from '../ImageWithCard';

describe('ImageWithCard', () => {
  it('renders the component', () => {
    const wrapper = shallow(<ImageWithCard src="" />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
