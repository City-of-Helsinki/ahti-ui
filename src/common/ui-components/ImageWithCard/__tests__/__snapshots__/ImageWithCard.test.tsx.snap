// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWithCard renders the component 1`] = `"<div style=\\"background-image:url()\\"></div>"`;
